### DISCUSSION TITLE
How to use the rasn-mib crate

### DISCUSSION THREAD

**Discussion by mat813** [librasn/rasn#388] [Q&A]
I am writing a tool to synchronize our netbox content with information from devices via snmp so that netbox always reflect reality as most as possible, and it feels like I should be using rasn_mib for some things, but I cannot figure out how it should be used, or how I could use the smi::object_type! macro for types that are not in rasn_mib.

**Comment by XAMPPRocky** [comment]
Thank you for your question! The documentation can definitely be improved (PRs welcome!), you can see examples of the macro in the codebase. As for how to use it, you're expected to use it in combination with `rasn` to decode the data. It does not come with a client to retrieve the data (as part of the value is that it can be used in any context e.g. sans-io). Though if a sans-io client can be added, that would also be welcome 🙂 

https://github.com/librasn/rasn/blob/8b42b5407581a1d4073ee51a3341d38b231334b3/standards/mib/src/lib.rs#L41-L49

**Reply by mat813** [reply]
Ok, my question was mostly about how to use rasn-mib. The client (agent) part is ok.

After deserializing, I get a response with a `v2c::Message<Pdus>`, which has a `.data` that is a `v2::Response`, which encapsulates a `Pdu` which has a `VarBindList`, which has `VarBind`s, and in there is a `VarBindValue`. Using matches, I get the content out of it, like, to get an Integer, I can do a match on:

```rust
           match vbv {
                VarBindValue::Value(ObjectSyntax::Simple(SimpleSyntax::Integer(value))) => {
                    Ok(value),
                }
                _ => Err("not an Integer"),
            }
```

Now, how do I plug in the rasn-mib object types in there ?

**Reply by XAMPPRocky** [reply]
I believe you need to match on the OID of the object, and then it can be mapped to the MIB type. We probably could add some kind of SMI registry type that encapsulates that functionality.
